src/main/java/com/amazon/opendistroforelasticsearch/sql/metrics/MetricFactory.java

```diff
@@ -0,0 +1,33 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.metrics;
+
+import com.amazon.opendistroforelasticsearch.sql.query.join.BackOffRetryStrategy;
+
+public class MetricFactory {
+
+    public static Metric createMetric(MetricType type) {
+        if (type.getType() == 1) {
+            return new NumericMetric<>(type.getName(), new RollingCounter());
+        } else if (type == MetricType.CIRCUIT_BREAKER) {
```

Comment:
it's a little confusing to have different cases based on different values - some for `type.getType()` and others for the `type` itself. It might introduce overlaps in the future, and generally it's hard to follow, please either use `type` everywhere, or `getType`.

Reply:
OK, will do it.